test/tests/pbs_Rrecord_resources_used.py

```diff
@@ -0,0 +1,616 @@
+
+        rc = self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA)
+        self.assertEqual(rc, 0)
+
+        rc = self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostB)
+        self.assertEqual(rc, 0)
+
+        a = {'resources_available.ncpus': 4}
+        rv = self.server.manager(MGR_CMD_SET, NODE, a, id=self.hostA, expect=True)
+        self.assertTrue(rv)
+        rv = self.server.manager(MGR_CMD_SET, NODE, a, id=self.hostB, expect=True)
+        self.assertTrue(rv)
+
+    def common(self, is_nonrerunnable, restart_mom):
+        """
+        Set node_fail_requeue=5 on server
+        """
+        a = {ATTR_nodefailrq: 5}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        """
+        Job script
+        """
+        test = []
+        test += ['#PBS -N RequeueTest\n']
+        test += ['#PBS -l ncpus=1\n']
+        test += ['echo Starting test at `date`\n']
+        test += ['sleep 1000\n']
+
+        test1 = []
+        test1 += ['#PBS -N RequeueTest\n']
+        test1 += ['#PBS -l nodes=1\n']
+        test1 += ['echo Starting test at `date`\n']
+        test1 += ['sleep 1000\n']
+
+        """
+        Submit three jobs J1,J2,J3[]
+        """
+        j1 = Job(TEST_USER, attrs={ATTR_k: 'oe'})
+        j1.create_script(body=test)
+        jid1 = self.server.submit(j1)
+
+        if is_nonrerunnable==True:
+            j2 = Job(TEST_USER, attrs={ATTR_r: 'n', ATTR_k: 'oe' })
+        else:
+            j2 = Job(TEST_USER, attrs={ATTR_r: 'y', ATTR_k: 'oe'})
+
+        j2.create_script(body=test1)
+        jid2 = self.server.submit(j2)
+
+        j3 = Job(TEST_USER, attrs={ATTR_J: '1-6:1', ATTR_k: 'oe'})
+        j3.create_script(body=test)
+        jid3 = self.server.submit(j3)
+
+        subjobs = self.server.status(JOB, id=jid3, extend='t')
+        jid3s1 = subjobs[1]['id']
+
+        """
+        Wait for the jobs to start running.
+        """
+        self.server.expect(JOB, {ATTR_state: 'R'}, jid1, offset=15)
+        self.server.expect(JOB, {ATTR_state: 'R'}, jid2, offset=15)
+        self.server.expect(JOB, {ATTR_state: 'R'}, jid3s1, offset=15)
+
+        """
+        Verify that accounting logs have Resource_List.<resource> value
+        """
+        m = self.server.accounting_match(
+        msg='.*Resource_List.*', id=jid1, regexp=True)
+        self.assertNotEqual(m, None)
+        m = self.server.accounting_match(
+        msg='.*Resource_List.*', id=jid2, regexp=True)
+        self.assertNotEqual(m, None)
+        m = self.server.accounting_match(
+        msg='.*Resource_List.*', id=jid3s1, regexp=True)
+        self.assertNotEqual(m, None)
+
+        """
+        Bring both moms down using kill -9 < mom pid>
+        """
+        self.momA.signal('-KILL')
+        self.momB.signal('-KILL')
+
+        """
+        Verify that both nodes are reported to be down.
+        """
+        self.server.expect(NODE, {ATTR_NODE_state: (
+        MATCH_RE, '.*down.*')}, id=self.hostA)
```

Comment:
Before TPP, it could take up to a full server ping cycle (5m) before a node would be marked down after the mom went away.  You may want to check with Subhasis if this changed with TPP.  If it hasn't, you may want to extend max_attempts and possibly interval.